File: engr/engineering/doc_events/additional_salary.py
import frappe
from frappe import _, bold




def on_validate(self,method):
    site_allow = frappe.db.get_value("Employee",self.employee,"site_allowances")
    if self.present_days_at_site:
        amount = (site_allow*self.present_days_at_site)
        self.amount = amount
   

File: engr/public/js/additional_salary.js
frappe.ui.form.on('Additional Salary',{
    present_days_at_site: function(frm){
		
		if (frm.doc.present_days_at_site){
			frappe.db.get_value("Employee",frm.doc.employee,"site_allowances",function(r){
				let amount = (r.site_allowances * frm.doc.present_days_at_site)
				frm.set_value("amount", amount);
			})
           
            
			
		}
	},

});

